plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -45,6 +48,8 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<Delete
         DeleteRequest deleteRequest = new DeleteRequest(ML_MODEL_INDEX, modelId);
 
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            Base64.getUrlDecoder().decode(modelId);
```

Comment:
Can you add comment: the OpenSearch core code to generate ID with `Base64.getUrlEncoder` ?

Reply:
One general suggestion is don't bind tightly with OpenSearch core code to avoid unnecessary effort if OpenSearch code changes in future.

Reply:
I think the simple check is ok, and ok if we don't check at all, the search request will always fail if pass invalid mode/task id.